### PR TITLE
Add cosine_embedding_loss to pytorch frontend

### PR DESCRIPTION
**Pull request by JSSan97**
cosine_embedding_loss: https://pytorch.org/docs/stable/generated/torch.nn.functional.cosine_embedding_loss.html
Subtask: https://github.com/unifyai/ivy/issues/10295
Issue: https://github.com/unifyai/ivy/issues/26

### CODE REVIEW

**Comment by JSSan97** [comment]
Hi there,

I'm ready for the first review. I'm able to get my tests working, I wrote three parts of tests to reflect the different target inputs: -1, 1 and 'anything else'. Some of the tests in the CI Pipeline seem to be unsuccessful but they seem to be tests related to other frontend loss functions?

Kind regards,

John

**Comment by JSSan97** [comment]
Hi @ZoeCD,

I know this might jump out of the scope of the coding challenge expectation, but I had a thought about whether the cosine similarity part of this loss function needs to be a compositional function (or mixed as there are keras and torch versions of cosine similarity), and if this is the case, a suggestion of where they should be.

**Comment by JSSan97** [comment]
I've fixed some logic and added more assertions since my last comment.
